pkg/reconciler/trigger/trigger.go

```diff
@@ -301,3 +365,25 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, t *v1alpha1.Trig
 	}
 	return newSub, nil
 }
+
+func getObjRefFromDependencyAnnotation(dependencyAnnotation string) (corev1.ObjectReference, error) {
+	var objectRef corev1.ObjectReference
+	if err := json.Unmarshal([]byte(dependencyAnnotation), &objectRef); err != nil {
+		return objectRef, err
+	}
+	return objectRef, nil
+}
+
+func validateDependencyAnnotation(depObjRef corev1.ObjectReference) error {
+	var result *multierror.Error
```

Comment:
You can probably replace this library with https://godoc.org/knative.dev/pkg/apis#FieldError. See https://github.com/knative/pkg/blob/0a0c4972cd7134ce54b1ce6a6de8f189257a64fb/apis/v1alpha1/destination.go#L148-L163 for an example of using it to validate an object reference.